src/business-logic/services/device-connection-service-implementation.ts

```diff
@@ -0,0 +1,106 @@
+import { DeviceConnectionService } from './interfaces/device-connection-service'
+import { Device } from '../../model/entities/device'
+import { DeviceConnectionFactory } from './interfaces/device-connection-factory'
+import { DeviceConnection } from './interfaces/device-connection'
+import { DeviceConnectionStatus } from '../../model/enums/device-connection-status'
+import { DeviceAlreadyConnectedException } from '../../model/exceptions/device-already-connected-exception'
+import { DeviceNotFoundException } from '../../model/exceptions/device-not-found-exception'
+import { Logger } from '@tv2media/logger/*'
+import { DeviceService } from './interfaces/device-service'
+
+type DeviceAggregate = {
+  deviceConnection: DeviceConnection,
+  device: Device
+}
+
+export class DeviceConnectionServiceImplementation implements DeviceConnectionService {
+  private readonly connectedDevices: Map<string, DeviceAggregate> = new Map()
+
+  constructor(private readonly deviceConnectionFactory: DeviceConnectionFactory, private readonly deviceService: DeviceService, private readonly logger: Logger) {
+    this.initialize(this.deviceService)
+      .then(() => {
+        logger.debug('Initialization of DeviceConnectionService done')
+      })
+      .catch((error) => {
+        logger.error(`Initialization of DeviceConnectionService failed with error: ${error}`)
+      })
+  }
+
+  private async initialize(deviceService: DeviceService): Promise<void> {
+    const devices: Device[] = await deviceService.getDevices()
+    await Promise.all(devices.map(device => this.createConnection(device)))
+  }
+
+  public async createConnection(device: Device): Promise<void> {
+    if (this.connectedDevices.has(device.id)) {
+      throw new DeviceAlreadyConnectedException(device.id)
+    }
+    const deviceConnection: DeviceConnection | undefined = this.deviceConnectionFactory.createDeviceConnection(device)
+
+    if(typeof deviceConnection === 'undefined'){
```

Comment:
`typeof` is not needed here. The condition can check against the `undefined` literal like `deviceConnection === undefined`.
Since the alternative to this being undefined is being a truthy value, and `undefined` is falsy, the condition can also be expressed as `!deviceConnection`.